GliaWidgets/SecureConversations/Confirmation/SecureConversations.ConfirmationViewController.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+
+extension SecureConversations {
+    final class ConfirmationViewController: ViewController {
+        var props: Props {
+            didSet {
+                guard props != oldValue else { return }
+                renderProps()
+            }
+        }
+
+        private let viewFactory: ViewFactory
+
+        init(
+            viewFactory: ViewFactory,
+            props: Props
+        ) {
+            self.viewFactory = viewFactory
+            self.props = props
+            super.init(nibName: nil, bundle: nil)
+        }
+
+        @available(*, unavailable)
+        required init?(coder: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+        }
+
+        override func loadView() {
+            super.loadView()
+            renderProps()
+        }
+
+        func renderProps() {
+            switch props {
+            case let .confirmation(welcomeProps):
+                renderSuccess(props: welcomeProps)
+            }
+        }
+
+        func renderSuccess(props: SecureConversations.ConfirmationView.Props) {
+            let confirmationView: ConfirmationView
+            if let currentView = view as? ConfirmationView {
+                confirmationView = currentView
+            } else {
+                confirmationView = viewFactory.makeSecureConversationsConfirmationView(props: props)
+                view = confirmationView
+            }
+            confirmationView.props = props
+        }
+    }
+}
+
+extension SecureConversations.ConfirmationViewController {
+    enum Props: Equatable {
+        case confirmation(SecureConversations.ConfirmationView.Props)
```

Comment:
Do we need Props as enum if it has one case? For Welcome screen it seemed reasonable to have enum because the plan was to a have more than one case. If there are no plans to add more cases, it seems reasonable to use struct instead.

Reply:
Like this? 1e4cc67bd0b6a0a86401bb7205fb13f904cfde12 It seems a little bit clumsy, with the view controller props just being an enclosure for the confirmation view props. Maybe I implemented it incorrectly.

Reply:
I guess if ConfirmationView contents would be moved to ConfirmationViewController then it would be less clumsy. However it seems like most of the views in the SDK take responsibility for rendering instead of view controllers, but it is up to you to decide if it is worth moving everything  to view controller.

Reply:
I think it is fine like that. It could be improved in terms of moving layout logic from ConfirmationView to ConfirmationViewController, but I'm not sure if it worth an effort.